pkg/webhook/admission/pod/logger_injector.go

```diff
@@ -0,0 +1,111 @@
+package pod
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"strings"
+)
+
+const (
+	LoggerContainerName    = "inferenceservice-logger"
+	LoggerConfigMapKeyName = "logger"
+	PodKnativeServiceLabel = "serving.knative.dev/service"
+)
+
+type LoggerConfig struct {
+	Image         string `json:"image"`
+	CpuRequest    string `json:"cpuRequest"`
+	CpuLimit      string `json:"cpuLimit"`
+	MemoryRequest string `json:"memoryRequest"`
+	MemoryLimit   string `json:"memoryLimit"`
+}
+
+type LoggerInjector struct {
+	config *LoggerConfig
+}
+
+func getLoggerConfigs(configMap *v1.ConfigMap) (*LoggerConfig, error) {
+
+	loggerConfig := &LoggerConfig{}
+	if loggerConfigValue, ok := configMap.Data[LoggerConfigMapKeyName]; ok {
+		err := json.Unmarshal([]byte(loggerConfigValue), &loggerConfig)
+		if err != nil {
+			panic(fmt.Errorf("Unable to unmarshall logger json string due to %v ", err))
+		}
+	}
+
+	//Ensure that we set proper values for CPU/Memory Limit/Request
+	resourceDefaults := []string{loggerConfig.MemoryRequest,
+		loggerConfig.MemoryLimit,
+		loggerConfig.CpuRequest,
+		loggerConfig.CpuLimit}
+	for _, key := range resourceDefaults {
+		_, err := resource.ParseQuantity(key)
+		if err != nil {
+			return loggerConfig, err
+		}
+	}
+
+	return loggerConfig, nil
+}
+
+func (il *LoggerInjector) InjectLogger(pod *v1.Pod) error {
+	// Only inject if the required annotations are set
+	_, ok := pod.ObjectMeta.Annotations[constants.LoggerInternalAnnotationKey]
```

Comment:
Instead of passing all of these through as annotations, can we simply read the pod's labels to get the inferenceservice and get the logging config from there? For this approach to be clean, we'd need the InferenceService's name, the component, and a label. We could then use our pkg/types code to have something like logger.GetContainerSpec the same way we do w/ other components. 

I'm trying to keep as much logic out of the webhook as possible, and encapsulate it within our types so that if we move away from a webhook approach, it doesn't take a big refactor.

WDYT?

Reply:
I think I would like to punt on this as it will add quite alot of complication to the mutator and general tests. Can we do this as an improvement later as we develop the use of the storage and logger mutators?